dht/i_partitioner.cc

```diff
@@ -179,6 +179,16 @@ unsigned shard_of(const schema& s, const token& t) {
     return s.get_sharder().shard_of(t);
 }
 
+static bool token_range_is_empty(const dht::token_range& r) {
+    return r.start()
+        && r.end()
+        && !r.start()->is_inclusive()
+        && !r.end()->is_inclusive()
+        && r.start()->value()._kind == dht::token_kind::key
+        && r.end()->value()._kind == dht::token_kind::key
+        && static_cast<uint64_t>(r.end()->value()._data) - static_cast<uint64_t>(r.start()->value()._data) == 1;
```

Comment:
I think it would be better to use https://github.com/scylladb/scylla/blob/1b46a0a5f114f50a295be388ceecd808283e87d5/dht/token.hh#L138